### PR TITLE
Feature/uwang 174

### PR DESCRIPTION
**Pull request by BilyHakim**
[FEATURE][UWANG-174] - membuat feature GetListUserByQuery

#### Description

> Description of features, bugs, improvements or any changes in this MR.

#### Ticket Reference(s)

(https://bluehabit.atlassian.net/browse/UWANG-174)

##### UI Changes

> Screenshot or video of UI-related changes

##### Coverage

> Test coverage screenshot.

N/A

#### Note(s)

> Mention other user or extra note unrelated to the MR here.

[UWANG-173]: https://bluehabit.atlassian.net/browse/UWANG-173?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[UWANG-174]: https://bluehabit.atlassian.net/browse/UWANG-174?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: routes/user/user.go [hunk 1]
```diff
@@ -34,3 +34,11 @@ func GetListUserWithPaginate(ctx *gin.Context) {
 	data := repo.GetListUser(page, size)
 	ctx.JSON(200, data)
 }
+
+func GetListUserWithQuery(ctx *gin.Context) {
+	query := ctx.DefaultQuery("query", "")
+
```
Comment (azkifairuz) [inline routes/user/user.go:45]: izin nambahin bang, biar aman dibikin valdasi buat cek kalo misal namenya atau querynya gada didatabse
```
query , ok := ctx.DefaultQuery("query", "")
if !ok {
		ctx.JSON(200, response.BadRequest("", "User tidak ditemukan"))
	}

```

**Review by azkifairuz — CHANGES_REQUESTED** [review]
punteng bang

File: repositories/user/user-repository.go [hunk 2]
```diff
@@ -77,6 +77,32 @@ func (repo *UserRespository) GetListUser(page string, size string) models.BaseRe
 
 }
 
+func (repo *UserRespository) GetListUserByQuery(query string) models.BaseResponse[[]models.UserCredentialResponse] {
+	var userCredential []entity.UserCredential
+	var userCredentialResponse []models.UserCredentialResponse
+	var response = models.BaseResponse[[]models.UserCredentialResponse]{}
+
+	err := repo.db.Where("name LIKE ?", "%"+query+"%").Find(&userCredential).Error
```
Comment (azkifairuz) [inline repositories/user/user-repository.go]: maaf bang, ditable userCredential gada name, adanaya full_name atau username. jdi nnti hasilnya "tidak menemukan data" terus karena dia g nemu fieldnya


**field yg ada **
```
type UserCredential struct {
	ID           string    `gorm:"column:id;primaryKey;default:8a4d80da-ec44-41db-b1ca-b7020d1418d4" json:"id"`
	Email        string    `gorm:"column:email;not null" json:"email"`
	Password     string    `gorm:"column:password;not null" json:"password"`
	Username     string    `gorm:"column:username;not null" json:"username"`
	FullName     string    `gorm:"column:full_name;not null" json:"full_name"`
	DateOfBirth  time.Time `gorm:"column:date_of_birth" json:"date_of_birth"`
	Gender       string    `gorm:"column:gender" json:"gender"`
	Status       string    `gorm:"column:status;not null" json:"status"`
	AuthProvider string    `gorm:"column:auth_provider;not null" json:"auth_provider"`
	CreatedAt    time.Time `gorm:"column:created_at;not null;default:CURRENT_TIMESTAMP" json:"created_at"`
	UpdatedAt    time.Time `gorm:"column:updated_at;not null;default:CURRENT_TIMESTAMP" json:"updated_at"`
	Deleted      bool      `gorm:"column:deleted;not null" json:"deleted"`
}
```

**Review by azkifairuz — CHANGES_REQUESTED** [review]

File: routes/user/user.go (hunk 1 above)
Comment (BilyHakim) [reply routes/user/user.go:45]: Oke good, udh gw tambahin ya

**Review by azkifairuz — APPROVED** [review]
kelass bang, tinggal nunggu bang trian acc

**Pull request merged**